server/src/main/java/org/opensearch/index/engine/EngineConfig.java

```diff
@@ -148,13 +151,42 @@ public Supplier<RetentionLeases> retentionLeasesSupplier() {
      * Compression Level gives a trade-off between compression ratio and speed. The higher compression level results in higher compression ratio but slower compression and decompression speeds.
      * This setting is <b>not</b> realtime updateable.
      */
-    public static final Setting<Integer> INDEX_CODEC_COMPRESSION_LEVEL_SETTING = Setting.intSetting(
+
+    public static final Setting<Integer> INDEX_CODEC_COMPRESSION_LEVEL_SETTING = new Setting<>(
```

Comment:
Looks like we allow customer to modify the compression level setting but simply ignore it unless its a zstd codec. 
Can we also have validation(`Setting.Validator`) to disallow user from setting the compression level based on `INDEX_CODEC_SETTING` value here?

Reply:
@ankitkala we don't need to add another validator since the validate() function defined below is taking care of it.